faros_event.sh

```diff
@@ -329,13 +343,168 @@ function processEventTypes() {
+                                    "run_end_time": $run_end_time,
+                                }'
+                            )
+                cicd_Build_with_start_end=$(concat "$cicd_Build_with_start_end" "$buildKey")
+                make_mutation cicd_build_with_start_end "$cicd_Build_with_start_end"
+            else
+                cicd_Build=$( jq -n \
+                                --arg run_status "$run_status" \
+                                --arg run_status_details "$run_status_details" \
+                                '{
+                                    "run_status": {"category": $run_status, "detail": $run_status_details},
+                                }'
+                            )
+                cicd_Build=$(concat "$cicd_Build" "$buildKey")
+                make_mutation cicd_build "$cicd_Build"
+            fi
+
+            cicd_Pipeline=$(jq \
+            '{data_run_pipeline,data_run_organization,data_run_source}' <<< $flat
+            )
+            make_mutation cicd_pipeline "$cicd_Pipeline"
+
+            cicd_Organization_from_run=$(jq \
+            '{data_run_organization,data_run_source}' <<< $flat
+            )
+            make_mutation cicd_organization_from_run "$cicd_Organization_from_run"
+        fi
+    else
+        make_mutation cicd_artifact "$artifact_key"
+    fi
+
+    cicd_ArtifactCommitAssociation=$(concat "$artifact_key" "$commit_key")
+    make_mutation cicd_artifact_commit_association "$cicd_ArtifactCommitAssociation"
+
+    cicd_Repository=$(jq \
+            '{data_artifact_repository,data_artifact_organization,data_artifact_source}' <<< $artifact_key
+            )
+    make_mutation cicd_repository "$cicd_Repository"
+
+    cicd_Organization=$(jq \
+            '{data_artifact_organization,data_artifact_source}' <<< $artifact_key
+            )
+    make_mutation cicd_organization "$cicd_Organization"
+}
+
+function make_mutation() {
+    log Calling Hasura rest endpoint $1 with payload $2
+
+    if !(($dry_run)); then
+        log "Sending mutation to Hasura..."
+
+        hasura_url="http://localhost:8080/api/rest"
+
+        http_response=$(curl --retry 5 --retry-delay 5 \
+            --silent --write-out "HTTPSTATUS:%{http_code}" -X POST \
+            "$hasura_url/$1" \
+            -H "content-type: application/json" \
+            -d "$2") 
+
+        http_response_status=$(echo $http_response | tr -d '\n' | sed -e 's/.*HTTPSTATUS://')
+        http_response_body=$(echo $http_response | sed -e 's/HTTPSTATUS\:.*//g')
+        if [ ! $http_response_status -eq 200 ]; then
+            err "[HTTP status: $http_response_status]"
+            err "Response Body:"
+            err "$http_response_body"
+            fail
+        else
+            log "[HTTP status OK: $http_response_status]"
+            log "Response Body:"
+            log "$http_response_body"
+        fi
+    else
+        log "Dry run: Event NOT sent to Faros."
```

Comment:
Maybe here say `Mutation` not sent to Faros?

Reply:
Good catch, thanks!